app/src/main/java/ch/hikemate/app/model/route/HikeRoutesRepositoryOverpass.kt

```diff
@@ -104,14 +107,102 @@ class HikeRoutesRepositoryOverpass(val client: OkHttpClient) : HikeRoutesReposit
       onSuccess(routes)
     }
 
+    /**
+     * Flattens a hike route into a list of points.
+     *
+     * This function is used to convert a hike route into a list of points that can be used to draw
+     * a polyline on the map. It also finds the starting and ending points of each sub way to create
+     * a smooth path.
+     *
+     * @param hikeWays The hike ways to be flattened.
+     * @return The list of points that represent the hike route. An empty list is returned either if
+     *   the hike is considered invalid or if there are no points in the hike.
+     */
+    private fun flattenHikeWays(hikeWays: List<HikeWay>): HikeWay {
+      val points = mutableListOf<LatLong>()
+      for (i in hikeWays.indices) {
+        if (points.isEmpty()) {
+          handleFirstWay(hikeWays, points, i)
+        } else {
+          handleSubsequentWays(hikeWays, points, i)
+        }
+      }
+      return points
+    }
+
+    /**
+     * Procedure to handles the first way of the hike during the flattening process.
```

Comment:
Thanks for the syntax error! I wrote "Handles..." first, but changed it after, that's the reason. For the word "Procedure", since the function doesn't return anything and only has side effect, it seems appropriated to me